include/boost/confoost/confoost.hpp

```diff
@@ -10,7 +10,170 @@
 #ifndef BOOST_CONFOOST_CONFOOST_HPP
 #define BOOST_CONFOOST_CONFOOST_HPP
 
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <vector>
+
+
+// TODO : implement this or find an alternative (tree nodes for our schema)
+struct confoost_node {
+    std::string id;
+    confoost_node** children;
+    confoost_node() {};
+};
+
+
+// This is what we discussed as common representation (schema whatever) of our configuration
+template<typename N>
+class confoost_config {
+private:
+    bool is_valid;
+    N* root;
```

Comment:
See the TODO comment above. Implementation of the tree as we discussed
today is arbitary. I needed a compiling code, hence the ptr. We should
discuss both type and membership properties of our tree tomorrow.

There is also a shared_ptr somewhere in the implementation to capture
cache. That seemed ugly to me (too), so that may use some improvements as
well.
On May 13, 2015 8:31 PM, "rmpowell77" <notifications@github.com> wrote:

> In include/boost/confoost/confoost.hpp
> <https://github.com/JeffGarland/liaw2015/pull/7#discussion_r30293795>:
>
> > +
> > +
> > +// TODO : implement this or find an alternative (tree nodes for our schema)
> > +struct confoost_node {
> > +    std::string id;
> > +    confoost_node** children;
> > +    confoost_node() {};
> > +};
> > +
> > +
> > +// This is what we discussed as common representation (schema whatever) of our configuration
> > +template<typename N>
> > +class confoost_config {
> > +private:
> > +    bool is_valid;
> > +    N* root;
>
> Raw Pointer?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JeffGarland/liaw2015/pull/7/files#r30293795>.
>